apps/verify/behaviours/validate-case-details.js

```diff
@@ -7,18 +7,12 @@ const baseUrl = `${config.saveService.host}:${config.saveService.port}/verify_lo
 
 module.exports = superclass => class extends superclass {
   async saveValues(req, res, next) {
-    let queryColumn;
-    switch(true) {
-      case !!req.form.values.brp:
-        queryColumn = 'brp';
-        break;
-      case !!req.form.values.uan:
-        queryColumn = 'uan';
-        break;
-      default:
-        return res.redirect('/incorrect-details-brp');
+    const queryColumn = req.sessionModel.get('sign-in-method') ?? null;
+    const queryValue = req.form.values[queryColumn].toUpperCase() ?? null;
```

Comment:
I would suggest to use destructuring values from req.forms for readability
`const { values } = req.form;`
and then use it later as ` values[queryColumn]` instead of `req.form.values[queryColumn]`

Reply:
Aslo add check before calling `.toUpperCase()` method

`const queryValue = values[queryColumn]?.toUpperCase() ?? null;`

Reply:
Added a check on `req.form.values[queryColumn]` before uppercasing but I'm not sure about destructuring `{ values }` in this case - would be a formal outlier imho.